src/dataflow-types/src/types.rs

```diff
@@ -1307,42 +1313,115 @@ pub mod sources {
+        pub region: Option<String>,
         /// The custom AWS endpoint to use, if any.
         pub endpoint: Option<SerdeUri>,
     }
 
-    /// Static AWS credentials for a source or sink.
+    /// AWS credentials for a source or sink.
     #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
-    pub struct AwsCredentials {
-        pub access_key_id: String,
-        pub secret_access_key: String,
-        pub session_token: Option<String>,
+    pub enum AwsCredentials {
+        /// Look for credentials using the [default provider][docs] chain
+        ///
+        /// [docs]: https://docs.rs/aws-config/latest/aws_config/default_provider/index.html
+        Default,
+        /// Load credentials using the given named profile
+        Profile { profile_name: String },
+        /// Use the credentials provided as WITH options
+        Static {
+            access_key_id: String,
+            secret_access_key: String,
+            session_token: Option<String>,
+        },
     }
 
     impl AwsConfig {
         /// Loads the AWS SDK configuration object from the environment, then
         /// applies the overrides from this object.
         pub async fn load(&self) -> mz_aws_util::config::AwsConfig {
-            use aws_smithy_http::endpoint::Endpoint;
-            use aws_types::credentials::SharedCredentialsProvider;
-            use aws_types::region::Region;
-
-            let mut loader = aws_config::from_env().region(Region::new(self.region.clone()));
-            if let Some(credentials) = &self.credentials {
-                loader = loader.credentials_provider(SharedCredentialsProvider::new(
-                    aws_types::Credentials::from_keys(
-                        &credentials.access_key_id,
-                        &credentials.secret_access_key,
-                        credentials.session_token.clone(),
-                    ),
-                ));
-            }
+            let mut loader: aws_config::ConfigLoader = aws_config::from_env();
+            // RegionProviderChain is not Clone so we must construct a new on for each use
+            let region = || {
+                if let Some(region) = &self.region {
+                    RegionProviderChain::first_try(Region::new(region.clone()))
+                        .or_default_provider()
+                } else {
+                    let mut wc =
+                        aws_config::default_provider::region::DefaultRegionChain::builder();
+                    if let AwsCredentials::Profile { profile_name } = &self.credentials {
+                        wc = wc.profile_name(profile_name);
+                    }
+                    // TODO: this doesn't work for profiles which do not provide their own region
+                    // but instead specify a source_profile that does have a region
+                    RegionProviderChain::first_try(wc.build()).or_default_provider()
+                }
+            };
```

Comment:
There's some nested provider chains going on here that seem weird. I think it's simpler to just return `Box<dyn ProvideRegion>` here to return the two types directly:

```rust
            let make_region_provider = || match &self.region {
                Some(region) => Box::new(Region::new(region.clone())) as Box<dyn ProvideRegion>,
                _ => {
                    let mut rc = DefaultRegionChain::builder();
                    if let AwsCredentials::Profile { profile_name } = &self.credentials {
                        rc = rc.profile_name(profile_name);
                    }
                    // TODO: this doesn't work for profiles which do not provide their own region
                    // but instead specify a source_profile that does have a region
                    Box::new(rc.build())
                }
            };
```

Reply:
Actually, I looked into what the SDK does internally, and you might as well just load the region here. It's much simpler that way:

```rust
            let region = match &self.region {
                Some(region) => Some(Region::new(region.clone())),
                _ => {
                    let mut rc = DefaultRegionChain::builder();
                    if let AwsCredentials::Profile { profile_name } = &self.credentials {
                        rc = rc.profile_name(profile_name);
                    }
                    // TODO: this doesn't work for profiles which do not provide their own region
                    // but instead specify a source_profile that does have a region
                    rc.build().region().await
                }
            };
```